Python/initconfig.c

```diff
@@ -1128,6 +1127,7 @@ _PyConfig_AsDict(const PyConfig *config)
     SET_ITEM_INT(use_frozen_modules);
     SET_ITEM_INT(safe_path);
     SET_ITEM_INT(_is_python_build);
+    SET_ITEM_INT(int_max_str_digits);
```

Comment:
You must also update _PyConfig_FromDict() so it becomes possible to override the config with _testinternalcapi.get_config() + _testinternalcapi.set_config() ;-)

I checked: sys.flags.int_max_str_digits should already be updated when _testinternalcapi.set_config() is called, it should work if you update _PyConfig_FromDict().